### ISSUE TITLE
Low Person Class Confidence 

### ISSUE THREAD

**Issue by miooooooooooo** [dbolya/yolact#72]
When I use CityScape dataset to train,now the mask mAP is about 13,but I find that the person class confidence is very low,and the average of it is only about 0.1,so What is the reason for this?

**Comment by dbolya** [comment]
Perhaps it's not trained enough, or maybe identifying people is really hard? I know in a preliminary test on cityscapes we were able to get 16-17 mAP with the base model, so it's probably the former, but cityscapes is a tough dataset if you don't use the full 1080p images (since without that, the instances are all really small and there are a lot of them).

**Comment by miooooooooooo** [comment]
Thank you for answer this question,maybe the parameter pred_aspect_ratios can cause this problem,because when I resize the image to 550*550,the person in the image has low width/height in common,so I try to change the pred_aspect_ratios value to (1/2,1/4,1).But I don't know whether there is a good result for person class

**Comment by dbolya** [comment]
Oh yeah that's a good idea, and I think it would work. However, YOLACTv1 has a bug where all the anchor boxes are squares (I accidentally made the width and height equal). If you're on my latest / close to latest commit, you can fix this by setting `'use_square_anchors': False` in the config's backbone. I have that set to true for backward compatibility.

**Comment by miooooooooooo** [comment]
ok,I will try it

**Comment by miooooooooooo** [comment]
I have tried it for 20000+ iterations and I set the batch_size=16,I find that the loss is lower but mAP is also more lower compare to before at a same time,but the difference is only 'parameter pred_aspect_ratios' value.So,What is the reason?

**Comment by dbolya** [comment]
So it's worse if you set `'use_square_anchors'` to false? Maybe it's overfitting? You can also try making the anchor scales smaller, e.g. instead of starting at 24, start at 16 and increase by 2x each time.

**Comment by miooooooooooo** [comment]
Thank you for your reply,I will try it.

**Comment by viet2411** [comment]
@miooooooooooo I'm trying to retrain this on Cityscape dataset. I'm using the code in facebook reasearch team's Detectron Github https://github.com/facebookresearch/Detectron/blob/master/tools/convert_cityscapes_to_coco.py
but this is not working. Can you tell me the way you create Cityscape dataset for yolact please?

**Comment by miooooooooooo** [comment]
oh,I don't use the "convert_cityscape_to_coco.py",I write the code which is very simple,just preserve the annotation about the polgon,bounding box,class and img_id.you can flow the coco annotation and write it quickily.

**Comment by dbolya** [comment]
Since this has been open so long, I'm going to close it. Feel free to reopen if you have any updates.

**Comment by areebsyed** [comment]
HI I want to detect persons only. With everything remaining the same what should I do to nullify all the other classes from being detected but persons only. I understand I wouldnt have to make a custom config for this. I tried to change yolact/layers/output_utils.py but it gives out error when theres a frame with no person.
